### PR TITLE
Add remove task

### PR DESCRIPTION
**Pull request by AbuRayhaan**
**To Do List: Add & Remove Tasks**

In this milestone, I did the following:
- Implement the `Add a new task` functionality.
- Implement the `Delete a new task` functionality.
- Implement the `Edit a new task` functionality.
- Implement storing changes to `local storage`.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,43 +1,192 @@
...
+addTask.addEventListener('click', (addItem) => {
+  addItem.preventDefault();
+  if (!addTaskInput.value) return;
+
+  if (editTaskItem != null) {
+    saveEditTask(editTaskItem);
+    editTaskItem = null;
+  } else {
+    saveTask({ index: storedTasks.length, description: addTaskInput.value, completed: false });
+  }
```
Comment (bini-i) [inline src/index.js]: - [ ] Kindly make sure the property index is set to the value of the new array length as described in the requirements(i.e. if a 5th task is added to the list, the index of that task should equal to the length of the new array (5)). 
- [ ] Currently a new task added gets an index of one less than the length of the new array.

File: src/index.js [hunk 2]
```diff
@@ -1,43 +1,192 @@
...
+    check.addEventListener('click', () => {
+      if (TasksR[i].completed === true) {
+        TasksR[i].completed = false;
+        checkBoxs[i].checked = false;
+        document.getElementById(`pTask${i}`).style.textDecoration = 'none';
+        localStorage.setItem('tasks', JSON.stringify(TasksR));
+      } else {
+        TasksR[i].completed = true;
+        checkBoxs[i].checked = true;
+        document.getElementById(`ptask${i}`).style.textDecoration = 'line-through rgb(68, 68, 68)';
```
Comment (bini-i) [inline src/index.js]: - [ ] Trying to check a task as completed throws an error in the console as shown below
image
- [ ] Kindly try to fix the typo `ptask`, the element you are trying to target with `document.getElementById('ptask${i}')` has an id of `pTask*`

File: src/index.js [hunk 3]
```diff
@@ -1,43 +1,192 @@
...
+  }
+});
+
+const SteerChecked = () => {
+  const StoreCheck = JSON.parse(localStorage.getItem('tasks'));
+  const checkBoxs = document.querySelectorAll('.check');
+  StoreCheck.forEach((a, i) => {
+    if (a.completed === true) {
+      checkBoxs[i].checked = true;
+      document.getElementById(`ptask${i}`).style.textDecoration = 'line-through rgb(68, 68, 68)';
```
Comment (bini-i) [inline src/index.js]: - [ ] Do the same here
- [ ] Kindly try to fix the typo `ptask`, the element you are trying to target with `document.getElementById('ptask${i}')` has an id of `pTask*`

File: src/index.js [hunk 4]
```diff
@@ -1,43 +1,192 @@
...
+  inputs.forEach((element) => {
+    element.addEventListener('keypress', (event) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        ArrayStoredParse.forEach((a, i) => {
+          const listUpdate = document.getElementById(`updt${i}`);
+          ArrayStoredParse[i].description = listUpdate.value;
+          localStorage.setItem('tasks', JSON.stringify(ArrayStoredParse));
+          window.location.reload();
+        });
```
Comment (bini-i) [inline src/index.js]: - [ ] Trying to edit a task description, results in deleting all the other descriptions as shown below: 
image

-**Hint**: As you can see in your code, for every edit task triggered,  line 176-180 is iterating through all the tasks description and sets the value to `listUpdate.value`(which is null as the other tasks are not being modified).

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle:            
### Hello @AbuRayhaan  :wave: 

Your project looks good and shows you have put a lot of effort into it. Kindly work on the following suggestions to further improve your project.
## To highlight
- Good job correctly setting up linters and Githubflow
- The project renders beautifully.
- Adding a new task and deleting tasks works perfectly. 

## To Improve
- Kindly check inline comments below :point_down: 

## Optional suggestions
-   [_OPTIONAL_] It would be nice to organize the methods in to different file so that you and others can easily navigate through your code base. Also helps you to easily map the logic behind your code and avoid bugs.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

#### If you have any questions feel free to ask within the PR thread.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (Code Reviewer)
> GitHub: [@bini-i](https://github.com/bini-i) 
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

File: src/index.js (hunk 1 above)
Comment (mahbubul14) [reply src/index.js]: > * [ ]  Kindly make sure the property index is set to the value of the new array length as described in the requirements(i.e. if a 5th task is added to the list, the index of that task should equal to the length of the new array (5)).
> * [ ]  Currently a new task added gets an index of one less than the length of the new array.

As the previous reviewer suggested this change, Please fix it.

**Here is the project requirement:**
r1

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @AbuRayhaan  ,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

# Required Changes ♻️

- [ ] Please check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linters error.
:heavy_check_mark: Nice PR description.
:heavy_check_mark: Clean code.

### Optional suggestions

- [ ] The commit message should be written in the imperative tense. `Ex: "Fix bug" instead of "Fixed bug".` [For more details.](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AbuRayhaan,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**